feat: create ipynb notebook for coverage checker

bodom0015 opened the pull request:
## Problem
We would like to create an easier interface for staff to work with the Coverage Checker

## Approach
* feat: create an ipynb for running the Coverage Checker

## How to Test
We offer two methods for using this notebook, along with a test case for the notebook itself (same for either application)

### ReadOnly View
1. Navigate to [this notebook in GitHub](https://github.com/healthyregions/SDOHPlace-MetadataManager/blob/lambert8/feature/notebook-for-coverage-checker/manager/coverage/SDOHCoverage.ipynb)
    * You should see that GitHub renders a read-only preview of the contents of the notebook, including previously-executed cells

### Setup: Google Colab (online)
1. Navigate to https://colab.research.google.com/
    * You should see the Open Notebook dialog appears in the center of the screen
3. Choose GitHub on the left side and enter this repository & branch name
    * `repository` = `https://github.com/healthyregions/SDOHPlace-MetadataManager`
    * `branch` = `lambert8/feature/notebook-for-coverage-checker`
    * You should see the `SDOHCoverage.ipynb` notebook is found and listed below
4. Choose the `manager/coverage/SDOHCoverage.ipynb` notebook below to import it into Google Colab
    * You should see the notebook is loaded and is now interactive
5. Run the test case below

### Setup: JupyterLab (local)
1. Change to the new `coverage` directory: `cd manager/coverage/`
2. Run the JupyterLab container: `docker compose up -d`
3. Navigate to http://localhost:8888
    * Login with token = `sdoh`
4. Run the test case below
6. Shutdown the JupyterLab container: `docker compose down`

### Testing the Notebook Itself
1. Run one of the two setup methods above (or for funsies, try out both!)
2. Execute the first and second cells (Shift + Enter)
3. Download a CSV dataset from the [SDOH&Place Data folder in Box](https://uofi.app.box.com/s/fqtslnfkpmgi32pb1cah1eyvmimvp740)
    * For example: [SVI_2010_US.csv](https://uofi.app.box.com/s/fqtslnfkpmgi32pb1cah1eyvmimvp740/file/1765513583484)
4. Enter 3 user inputs into the widgets from the second cell's output
    * `input_file` = `SVI_2010_US.csv`
    * `geography` = `tract`
    * `id_field` = `HEROPID`
5. Execute the third cell to run the coverage checker on the inputs above
    * You should see that the highlight IDs are reported at the end - these are the IDs that are missing from the dataset, so we generate a string that will exclude them from the map

mradamcox reviewed (changes requested):
Very nice! Definitely on the right track. I was able to load it up in Colab and see how it all works. A couple of notes for the next set of edits:

- No need to mention Box or any special handling of files from there. From the user perspective the data input should just be "I have a CSV on my laptop, I want to run it through this checker". So make sure that workflow is clearly handled.
- The field name input should be completely open, no need to have (or suggest) just three choices. I think a default value of FIPS would be best, with the ability to manually enter anything random. We'll also update the wording around how people will use that, but that can come later.
- The output as a single line of comma separated values is fine. We can add some text at the end about how to use it. It's easy enough to select the entire line and then copy/paste into the manager app. We'll update the app to handle comma-separated values.

Other small thing:

- As the path to this `coverage.py` file has changed, update the path to that file in the section on running it independently at the bottom of the main repo README.
- For the new README here, can you add a little more instruction on how to load this notebook in Colab, I figured it out, but just a little more explanation would be good. (edit: Just adding the steps you have in this PR description would be plenty!)

bodom0015 commented:
Thank you, @mradamcox! I believe I have addressed these issues:
* Adjusted the path to `coverage.py` in main `README.md` file
* Added more details similar to the PR description in `manager/coverage/README.md`
* The field name input is just an open text box, I had just highlighted FIPS, GEOID, and HEROPID as possible examples because those are the ones I had encountered previously. I did go ahead and change the default to `FIPS` for this, but the user should be able to enter any value for `input_file` + `id_field` (only `geography` is a dropdown) 
* Updated text within the notebook and README pertaining to Box - we do still offer a download for a sample CSV, but the instructions should no longer point to this as a requirement and offer text that indicates that the user should upload their own CSV data

mradamcox reviewed (approved):
Great! Thanks for addressing those points. I'll merge this in now and I expect we will use it next week to do some more testing.
